### PR TITLE
pkg/kubelet/kuberuntime: Add userns tests for NamespacesForPod

### PR DESCRIPTION
**Pull request by rata**
As requested by @tallclair here:
	https://github.com/kubernetes/kubernetes/pull/123909/files#r1523599275

cc @giuseppe @AkihiroSuda @tallclair @mrunalp 

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Cover more code with tests

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/util/util_test.go [hunk 1]
```diff
@@ -221,11 +239,84 @@ func TestNamespacesForPod(t *testing.T) {
...
-			actual, err := NamespacesForPod(test.input, &kubecontainertest.FakeRuntimeHelper{}, nil)
-			require.NoError(t, err)
-			require.Equal(t, test.expected, actual)
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, pkgfeatures.UserNamespacesSupport, test.usernsEnabled)()
+
+			fakeRuntimeHelper := kubecontainertest.FakeRuntimeHelper{
+				RuntimeHandlers: test.runtimeHandlers,
+			}
+			actual, err := NamespacesForPod(test.input, &fakeRuntimeHelper, &fakeRuntimeHandlerResolver{})
+			if test.expErr {
```
Comment (kannon92) [inline pkg/kubelet/kuberuntime/util/util_test.go:314]: I'd probably add a bit more here. Can we verify the kind of errors we are getting?

**Comment by kannon92** [comment]
/priority important-soon
/triage accepted

Marking it as important soon as it was called out during the kep freeze. I think this change should be safe to include during test-freeze as it only impacts unit tests.

File: pkg/kubelet/kuberuntime/util/util_test.go (hunk 1 above)
Comment (rata) [reply pkg/kubelet/kuberuntime/util/util_test.go:314]: @kannon92 what do you mean with the kind of error?

What we get here is from a mock, so I don't think it is super useful to test that the mock error contains some specific string or something. That doesn't add any real value, IMHO.

Comment (kannon92) [reply pkg/kubelet/kuberuntime/util/util_test.go:314]: I see. I just meant that we verify what kind of error we get but I didn't realize that the errors are mocked errors.

/lgtm

Comment (rata) [reply pkg/kubelet/kuberuntime/util/util_test.go:314]: Thanks for the review!

**Comment by rata** [comment]
@mrunalp @SergeyKanzhelev friendly ping?

**Comment by rata** [comment]
@SergeyKanzhelev ping?

**Review by mrunalp — APPROVED** [review]

**Comment by rata** [comment]
`featuregatetesting.SetFeatureGateDuringTest` was changed since I wrote the tests. Now I've adapted it to the new signature, it works fine locally.

**Comment by rata** [comment]
@mrunalp all tests green again, can you lgtm and approve again? :)

**Comment by rata** [comment]
@SergeyKanzhelev maybe you can have a look? :)

**Pull request merged**